sycl/test-e2e/lit.cfg.py

```diff
@@ -441,6 +441,53 @@ def open_check_file(file_name):
     else:
         config.substitutions.append(("%cuda_options", ""))
 
+# Check for HIP SDK
+check_hip_file = "hip_include.cpp"
+with open_check_file(check_hip_file) as fp:
+    print(
+        textwrap.dedent(
+            """
+        #define __HIP_PLATFORM_AMD__ 1
+        #include <hip/hip_runtime.h>
+        int main() {  hipError_t r = hipInit(0); return r; }
+        """
+        ),
+        file=fp,
+    )
+config.hip_libs_dir = lit_config.params.get("hip_libs_dir", config.hip_libs_dir)
+config.hip_include = lit_config.params.get(
+    "hip_include",
+    (config.hip_include if config.hip_include else config.sycl_include),
+)
+
+hip_options = hip_options = (
+    (" -L" + config.hip_libs_dir if config.hip_libs_dir else "")
+    + " -lamdhip64 "
+    + " -I"
+    + config.hip_include
+)
+if cl_options:
+    hip_options = (
+        " "
+        + (
+            config.hip_libs_dir + "/amdhip64.lib "
+            if config.hip_libs_dir
+            else "amdhip64.lib"
+        )
+        + " /I"
+        + config.hip_include
+    )
+
+with test_env():
+    sp = subprocess.getstatusoutput(
+        config.dpcpp_compiler + " -fsycl  " + check_hip_file + hip_options
```

Comment:
Do we need `-fsycl` here?

Reply:
probably not but i just copied what we did for cuda, i can remove it from both if you think thats better

Reply:
No difference to me at all.